azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -191,19 +181,17 @@ def initialize(self):
                     log_cgroup_warning("The agent is within an unexpected slice: {0}".format(agent_slice))
                     return
 
+                if self.using_cgroup_v2():
```

Comment:
shouldn't this check be done as part of the "if cgroups supported" logic that starts at line 138?

Reply:
if we move it over there, careful with the reset at line 144, which also does a reload

Reply:
This v2 check needs to happen after we get the cgroup api instance on line 160. using_cgroup_v2() checks if self._cgroups_api is an instance of the v2 implementation 

I think this check can happen before we check which slice the agent is in

Reply:
we can't move to 144 line, we should check systemd system before we call cgroups api. Because those api calls based on systemd system. I think we can move to before agent slice but not 144

Reply:
the systemd check itself needs to be moved to the logic for "supported".

seems like we need some cleanup of the logic here. we can add a TODO comment here and address this on the next iteration

Reply:
we can combine all in place but series of steps will be same as now

Reply:
the concern is not about the steps, but about the organization of the code. 

we do not support v2; we do not support systems on which systemd is not running; those conditions should be part of "is supported" logic

if you feel ok making those changes now, great. otherwise it can be marked to do later

Reply:
actually, this a hotfix for the release that is blocked. let's postpone those changes (but mark them as todo)